### PR TITLE
UCT/IB: Use ibv_create_ah() to test validity of a local gid

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: test/gtest/uct/ib/test_ib.cc [hunk 1]
```diff
@@ -261,21 +261,11 @@ class test_uct_ib_gid_idx : public test_uct_ib_with_specific_port {
...
-        ah_attr.port_num       = m_port;
-        ah_attr.is_global      = 1;
-        ah_attr.grh.dgid       = gid;
-        ah_attr.grh.sgid_index = gid_index;
-        ah_attr.grh.hop_limit  = 255;
-
-        struct ibv_ah *ah = ibv_create_ah(ib_md->pd, &ah_attr);
-        if (ah == NULL) {
-            UCS_TEST_SKIP_R("failed to create address handle on " + device_str.str());
+        if (!uct_ib_device_test_roce_gid_index(&ib_md->dev, m_port, &gid,
```
Comment (brminich) [inline test/gtest/uct/ib/test_ib.cc]: would not it fail on assert in `uct_ib_device_test_roce_gid_index`?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11993/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 27-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/2b763aeda6c4748fff9c3c839301de18a19494df/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/b68641aa9fb8f1a79c4b836cf5d8ef1c056c78cc/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/b0fa1b270301cbb8fc8d0305342968d98fcdbcf9/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/93de83d0a5d75159d4d9f40473b8dfe193e0ffdc/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/10b1af3764265fe501b81d801b7fdace5837643f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/97f326d43598f9c8fdca3c872399fa11f12216f6/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/62723f10a3d487daffc87b9517d1ec0fc635481a/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/93ba87adbddcf1e702b7b61d0b82805ec064ed33/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/d9948558bae325f839f345e858f3a456afdd55aa/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/574acc34e6b7a2eb1c49c7a1c0711120318e1c87/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/58526958a165cad48178cd036db77fcea210bf60/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/82293888102fea0228b99248c0cad8378fcd4adf/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/9811d4a0bfc443597524a3c7b77e5c11fc911348/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/3a1152a3fe6650bb2dd9b887dc1d5525a2b693ac/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/fdccfeb47128f7d1a4797401209ba771a7ef0892/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/8829cabb5314026aa5b95b34559ad21c5413c585/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/f45f2ebbf8313e60cbcbe3d995be5bb51ee57cce/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/24890a786e7ee1844a5438914278a871e5e3c810/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/95985b0d039eb3c0dff1b6dab2b3135b2e7b2feb/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/3eed43a2716e3642ef9e74b0f653ec58ff6bfcd9/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/4220d8ddfb1e9f4d81ad2deff94520578d7ef42c/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/ddc7fbca710727070fc36268094a32f13b12a0d9/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/c0dc13de2e8e2fa4a6135ebadbb44eee28c9c4ba/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/542a70ffa51f19ac12f2e5cbd355b24712ec2aab/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/95449e67b65cc22d4e1c421eff9cb630f5cb1f8b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/03b2988910228696005fe2c2c0507d9aed4a2c57/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/45c0832996dc6b17f2062516ee51749b1422dbdf/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/3d6d527fc8a31264dc4b5fd2cf75414440f3121d/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b0ec8df6306a093850274e85b7597ce3/raw/4ca12d10e405eeedd7db977eb57abf5e12c9d2e4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: test/gtest/uct/ib/test_ib.cc (hunk 1 above)
Comment (yosefe) [reply test/gtest/uct/ib/test_ib.cc]: the test checks port is roce: https://github.com/openucx/ucx/pull/4085/files/4bd1e44f12fee67fb4286036eb3e2caecb40492e#diff-f639bf578bf86fa86c73a758832ec754R230

**Comment by yosefe** [comment]
@brminich ok to merge?

**Pull request merged**